nucypher/blockchain/eth/actors.py

```diff
@@ -656,3 +696,342 @@ def request_evaluation(self, evidence):
+            raise self.ConfigurationError(message)
+        if not self.funding_eth:
+            raise self.ConfigurationError(f"{self.funding_account} has no ETH")
+
+        # Transfer NU
+        self.funding_power.activate()
+
+        # Send new staker account ETH
+        tx = {'to': staker.checksum_address, 'from': self.funding_account, 'value': self.eth_funding}
+        txhash = self.blockchain.client.send_transaction(transaction=tx)
+        _ether_transfer_receipt = self.blockchain.client.wait_for_receipt(txhash, timeout=120)  # TODO: Include in config?
+
+        # Send new staker account NU
+        _result = self.token_agent.transfer(amount=amount.to_nunits(),
+                                            sender_address=self.funding_account,
+                                            target_address=staker.checksum_address,
+                                            auto_approve=True)
+        return staker
+
+    def initialize_stake(self,
+                         amount: NU,
+                         duration: int,
+                         checksum_address: str = None,
+                         password: str = None,
+                         fund_now: bool = True
+                         ) -> Stake:
+
+        if password and not checksum_address:
+            staker = self.__create_staker(password=password)
+        else:
+            staker = Staker(is_me=True, checksum_address=checksum_address, blockchain=self.blockchain)
+
+        if fund_now:
+            self.__prepare_new_staker(staker=staker, amount=amount)
+
+        self.attach_transacting_power(checksum_address=staker.checksum_address, password=password)
+        new_stake = staker.initialize_stake(amount=amount, lock_periods=duration)
+        self.to_configuration_file(override=True)
+        return new_stake
+
+    def divide_stake(self,
+                     address: str,
+                     password: str,
+                     index: int,
+                     value: NU,
+                     duration: int):
+
+        staker = self.get_active_staker(address=address)
+        if not staker.is_staking:
+            raise Stake.StakingError(f"{staker.checksum_address} has no published stakes.")
+
+        self.attach_transacting_power(checksum_address=staker.checksum_address, password=password)
+        result = staker.divide_stake(stake_index=index,
+                                     additional_periods=duration,
+                                     target_value=value)
+        self.to_configuration_file(override=True)
+        return result
+
+    def sweep(self) -> dict:
+        """Collect all rewards from all staking accounts"""
+        receipts = dict()
+        for staker in self.stakers:
+            self.attach_transacting_power(checksum_address=staker.checksum_address)
+            receipt = self.collect_rewards(staker_address=staker.checksum_address)
```

Comment:
No password is passed to `self.attach_transacting_power()` and to `self.collect_rewards()`, which means this assumes that the transacting power is already cached with a password; otherwise, this won't work. Perhaps you can pass a list of passwords as input to `sweep()`? (I know I know.....yuck... 🤢 )

Reply:
Sweep is deprecated (for now) in `68426a32076f60be232a5b50b458f5da56d06f31`